modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -74,14 +77,7 @@ class S3BlobStore implements BlobStore {
     private final ThreadPool threadPool;
     private final Executor snapshotExecutor;
 
-    private final Stats stats = new Stats();
-
-    final RequestMetricCollector getMetricCollector;
-    final RequestMetricCollector listMetricCollector;
-    final RequestMetricCollector putMetricCollector;
-    final RequestMetricCollector multiPartUploadMetricCollector;
-    final RequestMetricCollector deleteMetricCollector;
-    final RequestMetricCollector abortPartUploadMetricCollector;
+    private final StatsCollectors statsCollectors = new StatsCollectors();
```

Comment:
Could we revert these changes to stats collection too? That way this PR is just about adding the `purpose` parameter to all the APIs, which is noisy but low-risk, and follow-ups which change the behaviour like here will be easier to review and will let us pin down any later problems with `git bisect`.

Reply:
Sure I reverted these changes as well. I believe this PR is now a pure refactor. Hence I adjusted the labels accordingly as well.